utils/constants.jsx

```diff
@@ -787,7 +787,9 @@ export const Constants = {
     },
 
     SPECIAL_MENTIONS: ['all', 'channel', 'here'],
+    SPECIAL_MENTIONS_REGEX: /\B@(channel|all|here)(?!(\.|-|_)*[^\W_])/gi,
```

Comment:
the `(?!(\.|-|_)*[^\W_])` part seems a bit complex, is it trying to say that only \w caracters are allowed? although testing it on an online regex does have a match with `@all---` which I'm not sure that's intended.

Reply:
That is intentional, the idea is that `@all` can be followed up by any number of `-` `.` `_`  as long as they are not followed by a word character (excluding underscores).

So the following should be valid according to this regex
`@here....._` 
`@all-----_` 
`@all_______` 
But the following should not be valid
`@all------s`
`@all_______s`
`@all.........s`

Is there a simpler way to achieve that last bit? this was the cleanest solution that I could come up with

Reply:
if that's intentional, that's good enough, then :D